src/authentication/config/authentication.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+class Authentication:
+
+    def __init__(self, cluster_configuration, service_configuration, default_service_configuration):
+        self.cluster_configuration = cluster_configuration
+        self.service_configuration = dict(default_service_configuration, **service_configuration)
+
+
+    def validation_pre(self):
+        print self.service_configuration["OIDC"]
```

Comment:
there's also `python3` in this change.
I suggest to make Python code compatible for both python 2 and 3, for newly added code currently, e.g.,
```py
from __future__ import print_function
print(self.service_configuration["OIDC"])
```

Reply:
I will remove this line. Its's for debugging.